devel/mimalloc/files/0001-prim.h-support-tls_slot-for-PPC.patch

```diff
@@ -0,0 +1,55 @@
+From 252ae136987b2c9933e28666f6c1494f6a255dd3 Mon Sep 17 00:00:00 2001
+From: barracuda156 <vital.had@gmail.com>
+Date: Fri, 7 Jul 2023 16:32:03 +0800
+Subject: [PATCH 1/2] prim.h: support tls_slot for PPC
+
+---
+ include/mimalloc/prim.h | 11 ++++++++++-
+ 1 file changed, 10 insertions(+), 1 deletion(-)
+
+diff --git include/mimalloc/prim.h include/mimalloc/prim.h
+index 9e56069..30be1af 100644
+--- include/mimalloc/prim.h
++++ include/mimalloc/prim.h
+@@ -8,6 +8,9 @@ terms of the MIT license. A copy of the license can be found in the file
+ #ifndef MIMALLOC_PRIM_H
+ #define MIMALLOC_PRIM_H
+ 
++#if defined(__APPLE__) && defined(__POWERPC__)
```

Comment:
only PowerPC systems need pthread.h? That would be extremely odd…

Reply:
Intel uses assembler there, I would assume it works regardless of SDK or compiler.

Reply:
Have you got any other systems to check these patches on? Given the systems you are interested in, you’d probably want to test them on Leopard i386, standard SnowLeopard Intel, and then run them past a current system.

You often try to make your patches specific to your systems of interest, but the kinds of patches you are doing most often don’t appear to me to be specific to gcc as a compiler or to powerpc as an architecture.

Reply:
I have no environment to test Intel builds, that too something like Leopard i386. I agree with you that here the patch isn’t GCC-specific, at least aside of `libatomic` part, but the choice is between fixing something that is actually tested and blind shot in a hope something else gets fixed along. I am fine with the latter though.

Reply:
There are other references to pthread in that file. pthread.h is included by internal.h, and so it should be picked up by prim.h.